manifests/candlepin.pp

```diff
@@ -55,8 +55,8 @@
     truststore_file              => $certs::candlepin::truststore,
     truststore_password          => $certs::candlepin::truststore_password,
     artemis_client_dn            => $artemis_client_dn,
-    java_home                    => '/usr/lib/jvm/jre-11',
-    java_package                 => 'java-11-openjdk',
+    java_home                    => '/usr/lib/jvm/jre-17',
+    java_package                 => 'java-17-openjdk',
```

Comment:
This is what I used to test: https://brewweb.engineering.redhat.com/brew/taskinfo?taskID=51553445

Reply:
That's a downstream only resources and in upstream we should not even link it. I've started a scratch build upstream using the srpm: http://koji.katello.org/koji/taskinfo?taskID=551900

Reply:
The result of installing that is that does pull in `java-17-openjdk-headless`, but also `java-1.8.0-openjdk-headless` and `java-1.8.0-openjdk-devel` so `/usr/bin/java` is still Java 1.8:
```console
# dnf module enable pki-core -y
# dnf install http://koji.katello.org/kojifiles/work/tasks/1901/551901/candlepin-4.3.1-1.el8.noarch.rpm -y
# rpm -q --whatrequires java-1.8.0-openjdk-headless
javapackages-tools-5.3.0-1.module_el8.0.0+11+5b8c10bd.noarch
jss-4.9.4-1.module_el8.7.0+1172+b9bb9c8d.x86_64
# java -version
openjdk version "1.8.0_362"
OpenJDK Runtime Environment (build 1.8.0_362-b08)
OpenJDK 64-Bit Server VM (build 25.362-b08, mixed mode)
```

This means setting `java_home` is certainly needed. Then setting `java_package` isn't a bad thing, since it ensures it's present.